test/unit/remote-config/remote-config-api-client.spec.ts

```diff
@@ -0,0 +1,161 @@
+ * limitations under the License.
+ */
+
+'use strict';
+
+import * as _ from 'lodash';
+import * as chai from 'chai';
+import * as sinon from 'sinon';
+import { RemoteConfigApiClient } from '../../../src/remote-config/remote-config-api-client';
+import { FirebaseRemoteConfigError } from '../../../src/remote-config/remote-config-utils';
+import { HttpClient } from '../../../src/utils/api-request';
+import * as utils from '../utils';
+import * as mocks from '../../resources/mocks';
+import { FirebaseAppError } from '../../../src/utils/error';
+import { FirebaseApp } from '../../../src/firebase-app';
+
+const expect = chai.expect;
+
+describe('RemoteConfigApiClient', () => {
+
+  const ERROR_RESPONSE = {
+    error: {
+      code: 404,
+      message: 'Requested entity not found',
+      status: 'NOT_FOUND',
+    },
+  };
+  const EXPECTED_HEADERS = {
+    'Authorization': 'Bearer mock-token',
+    'X-Firebase-Client': 'fire-admin-node/<XXX_SDK_VERSION_XXX>',
+    'Accept-Encoding': 'gzip',
+  };
+  const noProjectId = 'Failed to determine project ID. Initialize the SDK with service '
+    + 'account credentials, or set project ID as an app option. Alternatively, set the '
+    + 'GOOGLE_CLOUD_PROJECT environment variable.';
+
+  const mockOptions = {
+    credential: new mocks.MockCredential(),
+    projectId: 'test-project',
+  };
+
+  const clientWithoutProjectId = new RemoteConfigApiClient(
+    mocks.mockCredentialApp());
+
+  // Stubs used to simulate underlying api calls.
+  let stubs: sinon.SinonStub[] = [];
+  let app: FirebaseApp;
+  let apiClient: RemoteConfigApiClient;
+
+  beforeEach(() => {
+    app = mocks.appWithOptions(mockOptions);
+    apiClient = new RemoteConfigApiClient(app);
+  });
+
+  afterEach(() => {
+    _.forEach(stubs, (stub) => stub.restore());
+    stubs = [];
+    return app.delete();
+  });
+
+  describe('Constructor', () => {
+    it('should throw when the app is null', () => {
+      expect(() => new RemoteConfigApiClient(null as unknown as FirebaseApp))
+        .to.throw('First argument passed to admin.remoteConfig() must be a valid Firebase app instance.');
+    });
+  });
+
+  describe('getTemplate', () => {
+    it(`should reject when project id is not available`, () => {
+      return clientWithoutProjectId.getTemplate()
+        .should.eventually.be.rejectedWith(noProjectId);
+    });
+
+    it('should resolve with the requested template on success', () => {
+      const stub = sinon
+        .stub(HttpClient.prototype, 'send')
+        .resolves(utils.responseFrom({
+          conditions: [{ name: 'ios', expression: 'exp' }],
+          parameters: { param: { defaultValue: { value: 'true' } } },
+          version: {},
+        }, 200, { etag: 'etag-123456789012-1' }));
+      stubs.push(stub);
+      return apiClient.getTemplate()
+        .then((resp) => {
+          expect(resp.conditions).to.deep.equal([{ name: 'ios', expression: 'exp' }]);
```

Comment:
You can reduce some of this duplication if the test response was assigned to a variable. Then you can:

```
expect(resp.conditions).to.deep.equal(testResponse.conditions);
```